source/extensions/filters/http/jwt_authn/jwt_cache.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+#include <chrono>
+#include <memory>
+#include <string>
+
+#include "envoy/common/time.h"
+
+#include "absl/types/optional.h"
+#include "jwt_verify_lib/jwt.h"
+#include "jwt_verify_lib/verify.h"
+#include "simple_lru_cache/simple_lru_cache_inl.h"
+
+using ::google::simple_lru_cache::SimpleLRUCache;
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace JwtAuthn {
+
+// Cache key is the JWT string, value is parsed JWT struct.
+
+// The default number of entries in JWT cache is 100.
+constexpr int kJwtCacheSize = 100;
+
+class JwtCache {
+public:
+  JwtCache(int cache_size);
```

Comment:
take two cache arguments in the constructor:  cache_enabled, cache_size

use std::unique_ptr<SimpleLRUCache> for jwt_cache_.  since it may not be created if cache_enabled is false.

Reply:
Done.